api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -53,6 +53,17 @@ default String name() {
    */
   TableScan newScan();
 
+  /**
+   * Create a new {@link IncrementalAppendScan scan} for this table.
+   * <p>
+   * Once a scan is created, it can be refined to project columns and filter data.
+   *
+   * @return an incremental scan for appends only snapshots
+   */
+  default IncrementalAppendScan newIncrementalAppendScan() {
```

Comment:
@flyrain @aokolnychyi @openinx @chenjunjiedada @yittg @hameizi After discussing with @rdblue, we think it is probably cleaner to have separate newScan method for appends only and CDC read. 

In the future, we can add `Table#newIncrementalChangelogScan` and `IncrementalChangelogScan` interface